File: Javascript/algorithms/sorting/SelectionSort.js
const SelectionSort = (arr) => {
  for (let i = 0; i < arr.length - 1; i++) {
    let min = i
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[min]) {
        min = j
      }
    }
    if (i !== min) {
      //optimization
      // console.log(i, min)
      ;[arr[min], arr[i]] = [arr[i], arr[min]]
    }
  }
  return arr
}

console.log(SelectionSort([2, 1, 9, 76, 4]))

File: Java/algorithms/sorting/SelectionSort.java
package Java.algorithms.sorting;

import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] arr = {4, 6,1, 2, 10, 7, 8, 11, 3, 16};
        int[] result = selectionSort(arr);
        System.out.println(Arrays.toString(result));
    }

    public static int[] selectionSort(int[] arr){

        for(int i = 0 ; i < arr.length-1 ; i++){
            int min = i;
            for(int j = i+1 ; j < arr.length ; j++){
                if(arr[j] < arr[min]) {
                    min = j;
                }
            }
            int temp = arr[min];
            arr[min] = arr[i];
            arr[i] = temp;
        }
        return arr;
    }
}

